### ISSUE TITLE
[Enhancement CalDAV/CardDAV] Send Content-Location when handling collections without trailing slashes

### ISSUE THREAD

**Issue by rfc2822** [owncloud/core#9730]
When sending a PROPFIND request to the `/remote.php/caldav` (without trailing slash) collection, OwnCloud (correctly) responds with the `/remote.php/caldav/` (with slash) collection in the multi-status reply. This may be hard to parse/understand for CalDAV/CardDAV clients.

[RFC 4918 (WebDAV) 5.2 Collection Resources](http://tools.ietf.org/html/rfc4918#section-5.2) states: »There is a standing convention that when a collection is referred to by its name without a trailing slash, the server MAY handle the request as if the trailing slash were present.  In this case, it SHOULD return a Content-Location header in the response, pointing to the URL ending with the "/".«

So I suggest to add the Content-Location header when OwnCloud implicitly appends a slash to the requested URL.

**Comment by LukasReschke** [comment]
9.0 has a completely new DAV backend. Closing thus.
